src/main/java/com/google/jenkins/plugins/storage/AbstractBucketLifecycleManager.java

```diff
@@ -62,13 +62,14 @@
   /**
    * Constructs the base bucket OLM plugin from the bucket name and module.
    *
-   * NOTE: this hides the "make public" and "for failed" options of the
-   * base, which are mainly relevant for true uploads.
+   * NOTE: this hides the "make public", "for failed" and "strip path prefix"
+   * options of the base, which are mainly relevant for true uploads.
    */
   public AbstractBucketLifecycleManager(String bucket,
       @Nullable UploadModule module) {
     super(bucket, false /*sharedPublicly*/,
-        false /*forFailedJobs*/, module);
+        false /*forFailedJobs*/, false /*stripPathPrefix */,
+        null /*pathPrefix */, module);
```

Comment:
Can we combine 'pathPrefix' and 'stripPathPrefix' and only strip when 'pathPrefix' isn't null?